src/main/java/io/cryostat/net/web/http/api/v2/TargetSnapshotPostHandler.java

```diff
@@ -145,10 +156,45 @@ public boolean isAsync() {
                                     webServer.get().getDownloadURL(connection, rename),
                                     webServer.get().getReportURL(connection, rename));
                         });
-        return new IntermediateResponse<HyperlinkedSerializableRecordingDescriptor>()
-                .statusCode(201)
-                .addHeader(HttpHeaders.LOCATION, desc.getDownloadUrl())
-                .body(desc);
+
+        String snapshotName = desc.getName();
+        Optional<InputStream> snapshotOptional =
+                recordingTargetHelper.getRecording(connectionDescriptor, snapshotName).get();
+        if (snapshotOptional.isEmpty()) {
+            throw new ApiException(
+                    500,
+                    String.format(
+                            "Successful creation verification of snapshot %s failed",
+                            snapshotName));
+        } else if (snapshotIsEmpty(snapshotOptional.get())) {
+            recordingTargetHelper.deleteRecording(connectionDescriptor, snapshotName).get();
```

Comment:
Seeing this duplicated logic between the v1 and v2 handlers, I'm thinking there is now enough repeated code here that it's worth extracting and reusing. The `executeConnectedTask` block, `optional.isEmpty()` check, and `snapshotIsEmpty()` check together look like they belong together and could be extracted into some method like `RecordingTargetHelper#createSnapshot()`. Both API handlers (and the `RuleProcessor`) could then just call that method and always implement the same behaviour.